feathr_project/feathr/spark_provider/_synapse_submission.py

```diff
@@ -473,4 +497,32 @@ def _download_file_list(self, local_paths: List[str], result_paths, directory_cl
                 local_file.write(downloaded_bytes)
                 local_file.close()
             except Exception as e:
-                logger.error(e)
+                logger.error(e)       
+                
+    def _dir_exists(self, dir_path:str):
+        '''
+        Check if a directory in datalake already exists. Will also return the directory client
+        '''
+        datalake_path_split = list(filter(None, re.split('/|@', dir_path)))
+        if len(datalake_path_split) <= 3:
+            raise RuntimeError("Invalid directory path for datalake: {dir_path}")
+        dir_client = self.file_system_client.get_directory_client(
+                '/'.join(datalake_path_split[3:]))
+        return [dir_client, dir_client.exists()]
+    
+    def copy_files(self, source_path: str, target_path: str, over_write: bool = False):
```

Comment:
maybe we can rename this to something like `copy_files_between_datalake` to make it more obvious?